Automattic/newspack-theme
# Refactor: Remove style pack code

### All Submissions:


### Changes proposed in this Pull Request:

The theme has had a warning about style packs being removed since late February (https://github.com/Automattic/newspack-theme/pull/799), or since the 1.0.0-alpha.33 (pre-1.0.0) release, so it seems like it should be safe to remove the style pack code.

**This PR makes the following updates:** 

* Removes the inc/stylepacks.php, /js/src/style-packs-customizer.js and classes/class-newspack-style-packs-core.php files.
* Updates functions.php to remove references to the above PHP files, and to remove functions to check for style packs, and warn about style packs being used. 
* Removes stylepack-specific Sass files (which just included imports for the related child theme sass files, to generate identical CSS for the style packs). 
* Updated the inc/typography.php and inc/color-patterns.php files to remove references to specific style packs, and to update parent-theme specific styles with a check for `! is_child_theme()` instead of checking for the default style pack.
* In each child theme, remove the customizer.php file that removes the Style Packs panel, and remove references to style packs in code comments, replacing them with 'child theme'
* Remove old style pack-specific styles for Newspack Nelson (ads-related), and Newspack Sacha (editor preview-related); neither style would have been working with the child themes, so will be added to them in separate PRs to keep this as simple as possible _(laughs nervously)_. 
* Updates the author-bio.php in the parent theme to remove checks for style-4. 
* Updates the build script to remove the style pack CSS files. 

### How to test the changes in this Pull Request:

These steps are a little all over the place, but they'll make sure to touch on all the areas of the themes that have been updated:

1. Apply PR and run `npm run build`.
2. If you aren't already, switch to the Newspack parent theme. 
3. Set some custom fonts; and make sure 'Use all-caps for accent text' is checked
4. Set custom colours, and a custom header colour -- both of these will help test to make sure changes to how the parent theme applies custom fonts/colours still work after being updated.
5. Test the following sections with the parent theme:
* Make sure the Tertiary navigation in the mobile menu doesn’t have a background:

![image](https://user-images.githubusercontent.com/177561/82840553-df28fb80-9e87-11ea-9609-c6d4d513d860.png)

* Make sure part of the archive page title (the 'Category', 'Tag' etc part), picks up your primary colour:

![image](https://user-images.githubusercontent.com/177561/82840583-f8ca4300-9e87-11ea-975b-b1d9f97e9ebd.png)

* Make sure the site header is picking up the background colour you specifically set for the header background: 

![image](https://user-images.githubusercontent.com/177561/82840772-9b82c180-9e88-11ea-9025-0f68c61fa478.png)

* Edit a page and add a Homepage Posts block; make sure the post meta, and section header are picking up your custom primary colour in the editor: 

![image](https://user-images.githubusercontent.com/177561/82840874-f9170e00-9e88-11ea-84e9-a88380c13be7.png)

* Confirm that the Section Header on the block is using all caps; publish the page and confirm the same is happening on the front-end:

![image](https://user-images.githubusercontent.com/177561/82840902-151aaf80-9e89-11ea-9751-c7ea185a2b4d.png)

6. Navigate to a single post displaying the author bio, and confirm that it displays correctly with no errors (some conditional code for style 4 was removed from the author bio):

![image](https://user-images.githubusercontent.com/177561/82841135-c7eb0d80-9e89-11ea-886a-61f108258eb4.png)

7. Turn off the Co-Authors Plus plugin if enabled, or turn it on if disabled; re-check the author bio and confirm that it's still displaying as expected.
8. Cycle through each of the child themes and confirm there are no errors on the front end and in the Customizer (where they've all had code changes). Make sure your customizations are moved over.
9. Run `npm run release:archive`; confirm that there aren't any errors, and that the releases are generated as expected

### Other information:
